src/web/utils/setNativeProps.ts

```diff
@@ -0,0 +1,60 @@
+import { BaseProps, CreateComponentProps } from '../../types';
+import { camelCaseToDashed } from './utils';
+import { convertInt32ColorToRGBA } from './convertInt32Color';
+
+export const setNativeProps = (
+  nativeProps: BaseProps,
+  elementRef: React.MutableRefObject<SVGElement | null>,
+  lastMergedProps: React.MutableRefObject<Partial<CreateComponentProps>>
+) => {
+  const { style, ...rest } = nativeProps;
+  if (elementRef.current) {
+    const current = elementRef.current;
+
+    // Set attributes and styles
+    Object.keys(rest).forEach((cleanAttribute) => {
+      const cleanValue = rest[cleanAttribute as keyof typeof rest];
+      switch (cleanAttribute) {
+        case 'ref':
+        case 'children':
+          break;
+        case 'style':
+          Object.assign(current.style, cleanValue);
+          break;
+        case 'fill':
+          if (cleanValue && typeof cleanValue === 'object') {
+            const value = cleanValue as { payload: number };
+            current.setAttribute(
+              'fill',
+              convertInt32ColorToRGBA(value.payload)
```

Comment:
Why not 

```suggestion
            current.setAttribute(
              'fill',
              convertInt32ColorToRGBA((cleanValue as { payload: number }).payload)
```

or define this type earlier and then cast?

The thing is, when `TypeScript` will be compiled into js, it will result in the following line:

```js
const value = cleanValue;
```

which doesn't make much sense.

Reply:
Thanks ;)